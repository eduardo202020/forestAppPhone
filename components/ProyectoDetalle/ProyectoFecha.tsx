import React from 'react';
import { Text, StyleSheet } from 'react-native';
import { Calendar } from 'react-native-calendars';

interface ProyectoFechaProps {
  fecha: string;
}

export const ProyectoFecha: React.FC<ProyectoFechaProps> = ({ fecha }) => (
  <>
    <Text style={styles.title}>Fecha de Cultivo</Text>
    <Calendar
      current={fecha}
      markedDates={{
        [fecha]: { selected: true, selectedColor: '#2ecc71' },
      }}
      disabledByDefault
      hideExtraDays
      style={styles.calendar}
    />
  </>
);

const styles = StyleSheet.create({
  title: {
    fontWeight: 'bold',
    marginBottom: 4,
    marginTop: 8,
  },
  calendar: {
    marginBottom: 16,
  },
});
