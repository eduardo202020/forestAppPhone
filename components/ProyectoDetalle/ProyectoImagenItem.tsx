import React from 'react';
import { Image, StyleSheet, View } from 'react-native';

interface ProyectoImagenItemProps {
  uri: string;
}

export const ProyectoImagenItem: React.FC<ProyectoImagenItemProps> = ({
  uri,
}) => (
  <View style={styles.imageWrapper}>
    <Image source={{ uri }} style={styles.image} resizeMode="cover" />
  </View>
);

const styles = StyleSheet.create({
  imageWrapper: {
    marginRight: 12,
    borderRadius: 10,
    overflow: 'hidden',
    borderWidth: 1,
    borderColor: '#eee',
  },
  image: {
    width: 120,
    height: 80,
  },
});
