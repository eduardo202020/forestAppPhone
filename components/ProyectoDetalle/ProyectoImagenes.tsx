import React from 'react';
import { ScrollView, Image, StyleSheet, View, Text } from 'react-native';

interface ProyectoImagenesProps {
  imagenes: string[];
}

export const ProyectoImagenes: React.FC<ProyectoImagenesProps> = ({ imagenes }) => (
  <View>
    <Text style={styles.label}>Im√°genes</Text>
    <ScrollView horizontal style={styles.imagesRow}>
      {imagenes && imagenes.map((img, idx) => (
        <Image
          key={idx}
          source={{ uri: img }}
          style={styles.imageBox}
          resizeMode="cover"
        />
      ))}
    </ScrollView>
  </View>
);

const styles = StyleSheet.create({
  imagesRow: {
    flexDirection: 'row',
    gap: 8,
    marginBottom: 16,
  },
  imageBox: {
    width: 100,
    height: 100,
    backgroundColor: '#e0e0e0',
    borderRadius: 10,
    marginRight: 8,
  },
  label: {
    fontWeight: 'bold',
    marginBottom: 4,
    marginTop: 8,
  },
});
