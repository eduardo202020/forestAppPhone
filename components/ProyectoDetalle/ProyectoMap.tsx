import React from 'react';
import { StyleSheet } from 'react-native';
import MapView, { Marker } from 'react-native-maps';

interface ProyectoMapProps {
  location: { latitude: number; longitude: number };
}

export const ProyectoMap: React.FC<ProyectoMapProps> = ({ location }) => (
  <MapView
    style={styles.map}
    initialRegion={{
      latitude: location.latitude,
      longitude: location.longitude,
      latitudeDelta: 0.01,
      longitudeDelta: 0.01,
    }}
  >
    <Marker coordinate={location} />
  </MapView>
);

const styles = StyleSheet.create({
  map: {
    width: '100%',
    height: 180,
    borderRadius: 10,
    marginBottom: 16,
  },
});
