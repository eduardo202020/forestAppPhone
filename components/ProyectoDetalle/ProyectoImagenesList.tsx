import React from 'react';
import { View, FlatList, StyleSheet } from 'react-native';
import { ProyectoImagenItem } from './ProyectoImagenItem';

interface ProyectoImagenesListProps {
  imagenes: string[];
}

const ProyectoImagenesList: React.FC<ProyectoImagenesListProps> = ({
  imagenes,
}) => {
  if (!imagenes || imagenes.length === 0) return null;
  return (
    <View style={styles.container}>
      <FlatList
        data={imagenes}
        keyExtractor={(item, idx) => item + idx}
        renderItem={({ item }) => <ProyectoImagenItem uri={item} />}
        horizontal
        showsHorizontalScrollIndicator={false}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    marginVertical: 16,
  },
});

export default ProyectoImagenesList;
