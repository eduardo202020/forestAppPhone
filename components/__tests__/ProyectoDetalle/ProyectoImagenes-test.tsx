import React from 'react';
import renderer from 'react-test-renderer';
import { ProyectoImagenes } from '../../ProyectoDetalle/ProyectoImagenes';

describe('ProyectoImagenes', () => {
  it('renderiza correctamente una lista de imágenes', () => {
    const imagenes = ['https://img.com/1.png', 'https://img.com/2.png'];
    const tree = renderer
      .create(<ProyectoImagenes imagenes={imagenes} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renderiza correctamente sin imágenes', () => {
    const tree = renderer.create(<ProyectoImagenes imagenes={[]} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
