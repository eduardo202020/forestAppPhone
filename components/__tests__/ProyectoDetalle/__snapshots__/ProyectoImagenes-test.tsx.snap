// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProyectoImagenes renderiza correctamente sin im치genes 1`] = `
<View>
  <Text
    style={
      {
        "fontWeight": "bold",
        "marginBottom": 4,
        "marginTop": 8,
      }
    }
  >
    Im치genes
  </Text>
  <RCTScrollView
    horizontal={true}
    style={
      {
        "flexDirection": "row",
        "gap": 8,
        "marginBottom": 16,
      }
    }
  >
    <View />
  </RCTScrollView>
</View>
`;

exports[`ProyectoImagenes renderiza correctamente una lista de im치genes 1`] = `
<View>
  <Text
    style={
      {
        "fontWeight": "bold",
        "marginBottom": 4,
        "marginTop": 8,
      }
    }
  >
    Im치genes
  </Text>
  <RCTScrollView
    horizontal={true}
    style={
      {
        "flexDirection": "row",
        "gap": 8,
        "marginBottom": 16,
      }
    }
  >
    <View>
      <Image
        resizeMode="cover"
        source={
          {
            "uri": "https://img.com/1.png",
          }
        }
        style={
          {
            "backgroundColor": "#e0e0e0",
            "borderRadius": 10,
            "height": 100,
            "marginRight": 8,
            "width": 100,
          }
        }
      />
      <Image
        resizeMode="cover"
        source={
          {
            "uri": "https://img.com/2.png",
          }
        }
        style={
          {
            "backgroundColor": "#e0e0e0",
            "borderRadius": 10,
            "height": 100,
            "marginRight": 8,
            "width": 100,
          }
        }
      />
    </View>
  </RCTScrollView>
</View>
`;
